### PR TITLE
Quiddity, RequestId & Uuid deprecation

### LINKED ISSUE #27: Leverage Winter '21 new RequestId and Quiddity
Upgrade to Winter '21 (API v50.0) and either add these fields in addition to, or instead of, `Log__c.TransactionId__c`. If they can be used as replacement, then the class `Uuid` could probably be removed from the repo as well.

https://releasenotes.docs.salesforce.com/en-us/winter21/release-notes/rn_apex_Runtime_Detection.htm

### PR DESCRIPTION
**Pull request by jamessimone**
* Fixes #27 by taking advantage of Winter 21 features. 
* Removes Uuid dependency
* Small QOL logging improvements related to Quiddity granularity

### CODE REVIEW

**Comment by jamessimone** [comment]
@jongpie this can only be tested on sandboxes with Winter 21 till it's been fully released!

File: nebula-logger/main/callouts/classes/RestApi.cls [hunk 1]
```diff
@@ -5,7 +5,7 @@
 public with sharing class RestApi {
 
     private static Boolean ALL_OR_NONE_DEFAULT       = true;
-    private static String BASE_URL                   = System.Url.getSalesforceBaseUrl().toExternalForm() + '/services/data/v49.0';
+    private static String BASE_URL                   = System.Url.getSalesforceBaseUrl().toExternalForm() + '/services/data/v50.0';
```
Comment (jamessimone) [inline nebula-logger/main/callouts/classes/RestApi.cls:8]: I was considering making this programmatic a la the test - your thoughts?

**Comment by jongpie** [comment]
@jamessimone I added a couple of comments, but overall this looks great! Let's wait until next weekend to merge, once all environments have been upgraded to Winter '21.

File: sfdx-project.json [hunk 2]
```diff
@@ -2,10 +2,13 @@
     "packageDirectories": [
         {
             "path": "nebula-logger",
+            "package": "NebulaLogger",
```
[the remark this answers is not in the document]
Comment (jamessimone) [reply sfdx-project.json:5]: Makes sense, sure!

File: nebula-logger/main/logger/classes/Logger.cls [hunk 3]
```diff
@@ -4,10 +4,9 @@
 *************************************************************************************************/
 public without sharing class Logger {
 
-    public enum SaveMethod { DML, REST_API, QUEUEABLE_JOB }
-
+    private static final Request REQUEST = System.Request.getCurrent();
```
[the remark this answers is not in the document]
Comment (jamessimone) [reply nebula-logger/main/logger/classes/Logger.cls:7]: Yeah, I can remove the storage of it, I agree reducing heap size makes the slight repetition worth it

**Comment by jamessimone** [comment]
> @jamessimone I added a couple of comments, but overall this looks great! Let's wait until next weekend to merge, once all environments have been upgraded to Winter '21.

@jongpie sounds good. I rebased and added two commits, 5914f1cb51321703111bc0326b4f1806cacf4237 I think you'll like in particular for taking advantage of the safe navigation feature also new in Winter '21. It's possible that there are other instances in the codebase that can also take advantage of safe navigation to reduce boilerplate, as I haven't performed a complete scan.

I'll merge this PR next Sunday!

File: nebula-logger/main/logger/classes/LogEntryBuilder.cls [hunk 4]
```diff
@@ -171,25 +171,25 @@ public without sharing virtual class LogEntryBuilder {
         this.logEntryEvent.LimitsSoqlQueryRowsMax__c         = Limits.getLimitQueryRows();
         this.logEntryEvent.LimitsSoslSearchesMax__c          = Limits.getLimitSoslQueries();
         this.logEntryEvent.Locale__c                         = UserInfo.getLocale();
-        this.logEntryEvent.LoginDomain__c                    = sessionMap == null ? null : sessionMap.get('LoginDomain');
-        this.logEntryEvent.LoginHistoryId__c                 = sessionMap == null ? null : sessionMap.get('LoginHistoryId');
-        this.logEntryEvent.LoginType__c                      = sessionMap == null ? null : sessionMap.get('LoginType');
-        this.logEntryEvent.LogoutUrl__c                      = sessionMap == null ? null : sessionMap.get('LogoutUrl');
+        this.logEntryEvent.LoginDomain__c                    = sessionMap?.get('LoginDomain');
```
Comment (jongpie) [inline nebula-logger/main/logger/classes/LogEntryBuilder.cls:174]: I just realized you're using the new safe navigation now for null checks - love it!

Comment (jamessimone) [reply nebula-logger/main/logger/classes/LogEntryBuilder.cls:174]: it's a huge win for readability!

**Pull request merged**